definition_validator/validator.cc

```diff
@@ -280,10 +303,25 @@ class ValidateWalk {
         validateTStructNotGrandparent(ctx.state, sym);
         auto singleton = sym.data(ctx)->lookupSingletonClass(ctx);
         validateAbstract(ctx.state, singleton);
+        pushGenericContext(ctx, sym, singleton);
+        return classDef;
+    }
+
+    unique_ptr<ast::ClassDef> postTransformClassDef(core::Context ctx, unique_ptr<ast::ClassDef> classDef) {
+        popGenericContext();
         return classDef;
     }
 
     unique_ptr<ast::MethodDef> preTransformMethodDef(core::Context ctx, unique_ptr<ast::MethodDef> methodDef) {
+        // Only perform this check if this isn't a module from the stdlib, and
+        // if there are type members in the owning context.
+        auto methodData = methodDef->symbol.data(ctx);
+        auto isSelfMethod = methodData->owner.data(ctx)->isSingletonClass(ctx);
+        auto &frame = genericContext.back();
```

Comment:
can `frame` information can be extracted from `methodData.owner` making `frame` unnecessary?

Reply:
Yes. My goal was to cache these checks, as they're the same for each method of a single class. I'm happy to go back to the way it was before, which would do these checks on the method instead.

Reply:
Lets optimize for readability for now & thus have less state.
I expect that this code will be comparably fast and thus isn't worth optimizing much.